### PR TITLE
test(canvas): add canvas test case and fix the null pointer access crash in the draw buffer

### PR DESCRIPTION
**Pull request by vela-mib666**
Improve code coverage

```bash
================================================================
 Coverage analysis results for 'src/widgets/canvas/lv_canvas.c' 
================================================================
Coverable lines (per gcovr): 237
Covered lines: 213
Uncovered lines: 24
Coverage: 89.87%
```

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/widgets/canvas/lv_canvas.c [hunk 1]
```diff
@@ -280,18 +286,18 @@ const void * lv_canvas_get_buf(lv_obj_t * obj)
...
  *====================*/
 
-void lv_canvas_copy_buf(lv_obj_t * obj, const lv_area_t * canvas_area, lv_draw_buf_t * dest_buf,
-                        const lv_area_t * dest_area)
+void lv_canvas_copy_buf(lv_obj_t * obj, const lv_area_t * canvas_area, lv_draw_buf_t * src_buf,
+                        const lv_area_t * src_area)
 {
     LV_ASSERT_OBJ(obj, MY_CLASS);
-    LV_ASSERT_NULL(canvas_area && dest_buf);
+    LV_ASSERT_NULL(src_buf);
```
[the remark this answers is not in the document]
Comment (FASTSHIFT) [reply src/widgets/canvas/lv_canvas.c:293]: area supports passing NULL.

**Review by FASTSHIFT — APPROVED** [review]

**Review by XuNeo — APPROVED** [review]
